frontend/src/old-pages/Clusters/util.tsx

```diff
@@ -9,25 +9,61 @@
 // limitations under the License.
 // @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'js-y... Remove this comment to see the full error message
 import jsyaml from 'js-yaml'
+import {NavigateFunction} from 'react-router-dom'
 
-import {DescribeCluster, GetConfiguration} from '../../model'
 import {clearState, getState, setState} from '../../store'
+import {ClusterInfoSummary, ClusterStatus} from '../../types/clusters'
 import {getIn} from '../../util'
 
-export function selectCluster(clusterName: any, navigate: any) {
+export function canFetchClusterDetails(cluster?: ClusterInfoSummary) {
+  switch (cluster?.clusterStatus) {
+    case ClusterStatus.CreateInProgress:
```

Comment:
Question(possibly blocking): not all the states are here (I'm thinking about update_rollback_complete, update_rollback_in_progress and so on) are you sure we should return false in those cases?

Reply:
This part is a bit tricky, I want to start by "allowing" only the states that we know make sense. This way we have complete control and there is no risk of unwanted states resulting in a `DescribeCluster`. Here is the complete list of the supported clusterStatuses:

```ts
export enum ClusterStatus {
  CreateInProgress = 'CREATE_IN_PROGRESS',
  CreateFailed = 'CREATE_FAILED',
  CreateComplete = 'CREATE_COMPLETE',
  DeleteInProgress = 'DELETE_IN_PROGRESS',
  DeleteFailed = 'DELETE_FAILED',
  DeleteComplete = 'DELETE_COMPLETE',
  UpdateInProgress = 'UPDATE_IN_PROGRESS',
  UpdateComplete = 'UPDATE_COMPLETE',
  UpdateFailed = 'UPDATE_FAILED',
}
```

Do you see any state here that you would treat differently?
Thanks!

Reply:
Agreed offline to try to change the approach